ironfish-cli/src/commands/chain/foo.ts

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import {
+  Assert,
+  Block,
+  GENESIS_BLOCK_SEQUENCE,
+  Meter,
+  NodeUtils,
+  Spend,
+  TimeUtils,
+} from '@ironfish/sdk'
+import { CliUx, Flags } from '@oclif/core'
+import { IronfishCommand } from '../../command'
+import { LocalFlags } from '../../flags'
+import { ProgressBar } from '../../types'
+
+export default class Repair extends IronfishCommand {
+  static description = ''
+
+  static flags = {
+    ...LocalFlags,
+    dry: Flags.boolean({
+      default: false,
+      description: 'Dry run repair first',
+    }),
+    batchSize: Flags.integer({
+      default: 1000,
+      description: 'How many blocks to load in parallel',
+    }),
+    start: Flags.integer({
+      default: GENESIS_BLOCK_SEQUENCE,
+      description: 'Where to start repairing',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(Repair)
+
+    const node = await this.sdk.node()
+    await NodeUtils.waitForOpen(node)
+
+    let nullifierTreeIndex = 0
+    let nullifierRepaired = 0
+    let current = flags.start
+    const stop = node.chain.head.sequence
```

Comment:
Should this be the count of the nullifier tree? `node.chain.nullifiers.getCount('Leaves')`. If we are measuring the speed of how many nullifiers have been processed compared to the total number

Reply:
The progress is block based, but I do display speed of nullifier processing. I did display progress as nullifier # but, I want the ability to stop and resume from a certain block so this progress bar shows block progress now. I think user's will understand block progress a bit better too.